lib/domain/repository/chat.dart

```diff
@@ -119,6 +121,9 @@ abstract class AbstractChatRepository {
   /// Marks the specified [Chat] as hidden for the authenticated [MyUser].
   Future<void> hideChat(ChatId id);
 
+  /// Marks the specified [Chat] as archive for the authenticated [MyUser].
+  Future<void> toggleChatArchivation(ChatId id, bool archive);
```

Comment:
I suggest digesting from backend GraphQL mutation naming here, because "toggling archivation" sounds a bit too hard to understand. How about just `archiveChat(ChatId id, bool archive)`?